### ISSUE TITLE
How to stabilize training of keypoint detector?

### ISSUE THREAD

**Issue by kenmbkr** [AliaksandrSiarohin/first-order-model#161]
Due to random initialization, keypoints at the initial epochs can sometimes be evenly distributed or concentrated if I restart the experiment with the same configuration a few times.

1. Will different random initialization eventually converge to evenly distributed keypoints like the pre-trained models?
2. Are there better ways to initialize the keypoint detector so that the initial locations of the learned keypoints are evenly distributed?

**Comment by AliaksandrSiarohin** [comment]
1. It should, but may stuck in some local minimum. So beeter to initialize evenly distributed.
2. This would certainly help, but I don't know simple way to do it.

**Comment by SystemErrorWang** [comment]
@AliaksandrSiarohin 
About the keypoint detector, i would like to know if more keypoints or pre-trained keypoint detector model will improve the final results (in vox dataset)?
I tried to initilize the network with a pretrained face kp detector network with 36 points, but the network diverged after several epoches; I also tried to freeze the parameters of pretrained kp detector, but the loss will stuck at a high value. would like to know if you have tried similar methods, or if you have any idea about it.
Thank you very much for your nice work!

**Comment by AliaksandrSiarohin** [comment]
I did not try this for faces, I tried for Taichi both freezing and additional supervised Loss was working pretty good.
